integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java

```diff
@@ -0,0 +1,503 @@
+    templateMap.put("DOMAIN_NS", domainNamespace);
+    templateMap.put("DOMAIN_UID", domainUid);
+    templateMap.put("CLUSTER", clusterName);
+    templateMap.put("TLS_CERT", tlsSecretName);
+
+    Path srcVoyagerHttpsFile  = Paths.get(RESOURCE_DIR,
+        "tunneling", "voyager.tls.tunneling.template.yaml");
+    Path targetVoyagerHttpsFile = assertDoesNotThrow(
+        () -> generateFileFromTemplate(srcVoyagerHttpsFile.toString(),
+            "voyager.tls.tunneling.yaml", templateMap));
+    logger.info("Generated Voyager Https Tunneling file {0}", targetVoyagerHttpsFile);
+
+    createCertKeyFiles(K8S_NODEPORT_HOST);
+    createSecretWithTLSCertKey(tlsSecretName);
+    createJksStore();
+
+    StringBuffer deployTlsIngress = new StringBuffer("kubectl apply -f ");
+    deployTlsIngress.append(Paths.get(RESULTS_ROOT, "voyager.tls.tunneling.yaml"));
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(deployTlsIngress), true));
+
+    logger.info("kubectl apply returned {0}", result.toString());
+    checkServiceExists("voyager-voyager-tls-tunneling", domainNamespace);
+
+    int httpsTunnelingPort =
+        getServiceNodePort(domainNamespace, "voyager-voyager-tls-tunneling");
+    assertTrue(httpsTunnelingPort != -1,
+        "Could not get the HttpsTunnelingPort service node port");
+    logger.info("HttpsTunnelingPort for Voyager {0}", httpsTunnelingPort);
+
+    StringBuffer httpsUrl = new StringBuffer("https://");
+    httpsUrl.append(K8S_NODEPORT_HOST + ":" + httpsTunnelingPort);
+
+    StringBuffer javasCmd = new StringBuffer("java -cp ");
+    javasCmd.append(Paths.get(RESULTS_ROOT, "wlthint3client.jar"));
+    javasCmd.append(":");
+    javasCmd.append(Paths.get(RESULTS_ROOT));
+    // javasCmd.append(" -Djavax.net.debug=all ");
+    javasCmd.append(" -Djavax.net.ssl.trustStorePassword=password");
+    javasCmd.append(" -Djavax.net.ssl.trustStoreType=jks");
+    javasCmd.append(" -Djavax.net.ssl.trustStore=");
+    javasCmd.append(jksTrustFile);
+    javasCmd.append(" JmsTestClient ");
+    javasCmd.append(httpsUrl);
+    logger.info("java command to be run {0}", javasCmd.toString());
+
+    // Note it takes a couples of iterations before the client success
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Wait for Https JMS Client to access WLS "
+                    + "(elapsed time {0}ms, remaining time {1}ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(runJmsClient(new String(javasCmd)));
+  }
+
+  // Download the wlthint3client.jar from Adminserver pod to local filesystem.
+  // Use wlthint3client.jar in classpath to build and run the standalone 
+  // JMS client that sends messages to a Uniform Distributed Queue using 
+  // load balancer http(s) url which maps to custom channel on cluster member
+  // server on WebLogic cluster.
+  public void buildClient() {
```

Comment:
make this method as private or provide javadoc

Reply:
modified the scope to private